SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -0,0 +1,292 @@
+    _log "  -r, --req REQUIREMENTS      Path to requirements.txt file or http link for a public repository (mandatory)"
+    _log "  -c, --clear                 Clear the current virtual environment, if it exists"
+    _log "  -h, --help                  Print this help page"
+    _log "  --accpy VERSION             Version of Acc-Py to be used (options: ${ACCPY_ALL_VERSIONS_STR})"
+}
+
+# --------------------------------------------------------------------------------------------
+
+# Parse command line arguments
+while [ $# -gt 0 ]; do
+    key="$1"
+    case $key in
+        --env|-e)
+            ENV_NAME=$2
+            shift
+            shift
+            ;;
+        --req|-r)
+            REQUIREMENTS=$2
+            shift
+            shift
+            ;;
+        --clear|-c)
+            CLEAR_ENV=--clear
+            shift
+            ;;
+        --help|-h)
+            print_help
+            exit 0
+            ;;
+        --accpy)
+            ACCPY_CUSTOM_VERSION=$2
+            shift
+            shift
+            ;;
+        *)
+            >&2 _log "ERROR: Invalid argument: $1" && _log
+            print_help
+            exit 1
+            ;;
+    esac
+done
+
+# --------------------------------------------------------------------------------------------
+
+ENV_PATH="/home/$USER/${ENV_NAME}"
+
+# Checks if a name for the environment is given
+if [ -z "$ENV_NAME" ]; then
+    >&2 _log "ERROR: No virtual environment name provided." && _log
+    print_help
+    exit 1
+fi
+
+# Checks if an environment with the same name was already created, if --clear is not passed
+if [ -d "$ENV_PATH" ] && [ -z "$CLEAR_ENV" ]; then
+    >&2 _log "ERROR: Virtual environment already exists."
+    exit 1
+fi
+
+# Checks if the provided Acc-Py version is valid
+if [ -n "$ACCPY_CUSTOM_VERSION" ] && [ ! -e "$ACCPY_BASE/$ACCPY_CUSTOM_VERSION" ]; then
+    >&2 _log "ERROR: Invalid Acc-Py version. Options: ${ACCPY_ALL_VERSIONS_STR}"
+    exit 1
+fi
+
+# Checks if a requirements file is given
+if [ -z "$REQUIREMENTS" ]; then
+    >&2 _log "ERROR: No requirements provided." && _log
+    print_help
+    exit 1
+# Checks if the provided requirements source is found
+elif [ -f $REQUIREMENTS ]; then
+    if [[ ${REQUIREMENTS##*.} != "txt" ]]; then
+        >&2 _log "ERROR: Invalid requirements file."
+        exit 1
+    fi
+    REQ_PATH=$REQUIREMENTS
+elif [[ $REQUIREMENTS == http* ]]; then
+    # Extract the repository name from the URL
+    repo_name=$(basename $REQUIREMENTS)
+    repo_name=${repo_name%.*}
+    
+    mkdir -p $HOME/SWAN_projects
+    REPO_PATH=$HOME/SWAN_projects/${repo_name}
+
+    # Checks if the repository already exists in the home directory
+    if [ -d "${REPO_PATH}" ]; then
+        if [ -z "$CLEAR_ENV" ]; then
+            >&2 _log "ERROR: ${REPO_PATH} already exists. Use --clear to reclone the repository."
+            exit 1
+        else
+            rm -rf ${REPO_PATH}
```

Comment:
This is dangerous, because we are removing stuff from the user's CERNBox, I would not do it this way.
Counterproposal: we check if a directory with the same name exists under SWAN_projects. If that is the case, we append a number to the repo path (starting with 1); if it still exists, we try with 2, and so on. It's the same we do now when we clone git repos into SWAN projects.

Reply:
Currently, for implementing that mechanism, some (a lot) code had to be changed and not just there, because the API calling the script wouldn't be aware of that change in the name of the project_folder (which is the folder containing the requirements file). Which means it would be always accessing the first clone. I believe we could fix this if we did that checking in the API layer

Reply:
You mean checking in the Python backend of the SwanCustomEnv extension if, in case of EOS path, the path exists already and if so append a number already there?

In any case, I'm thinking we should remove the --clear option and the `rm` altogether, we don't want to end up removing stuff that the user didn't intend to remove. If a folder exists, we create another one. Period.